app/controllers/application_controller.rb

```diff
@@ -3,9 +3,18 @@ class ApplicationController < ActionController::Base
   before_action :http_basic_authenticate
 
   rescue_from ActionController::InvalidAuthenticityToken, with: :session_expired
+  rescue_from ActionController::RoutingError, with: :render_not_found
+
+  def raise_not_found
+    raise ActionController::RoutingError, "Not Found"
```

Comment:
Is it worth adding a bit more context to the raised error, like what's not been found? If it's already visible elsewhere then ignore this

Reply:
I think wherever this is used it should always be inferable/have enough context from the URL, so where I currently have it you would get a 404 on a path like `/teacher_training_adviser/sign_up/some-wrong-step`